Plugins/NativeEngine/Source/VertexBuffer.cpp

```diff
@@ -1,16 +1,22 @@
 #include "VertexBuffer.h"
 #include <cassert>
 
+#include <arcana/threading/task.h>
+#include <arcana/threading/task_schedulers.h>
+
 namespace
 {
-    template<typename sourceType> void PromoteToFloats(std::vector<uint8_t>& bytes, uint32_t numElements, uint32_t byteOffset, uint32_t byteStride)
+    template<typename sourceType>
+    void PromoteToFloats(gsl::span<uint8_t>& bytes, uint32_t numElements, uint32_t byteOffset, uint32_t byteStride)
     {
         const size_t count = bytes.size() / byteStride;
         const size_t destinationSize = count * numElements * sizeof(float);
-        if (destinationSize != bytes.size()) // ensure both vectors have different size
+        if (destinationSize != static_cast<size_t>(bytes.size())) // ensure both vectors have different size
         {
-            std::vector<uint8_t> newBytes(destinationSize);
-            float* destination = reinterpret_cast<float*>(newBytes.data());
+            uint8_t* newBytes = new uint8_t[destinationSize];
```

Comment:
Nit: I'd consider using a `std::unique_ptr<uint8_t[]>` here rather than a raw array to get the RAII cleanup and avoid the need for manual deletion a few lines down. In practice it doesn't really matter because I don't think any of these lines below can throw and thus leak it, but it still reads somewhat cleaner.

Reply:
I'd actually recommend continuing to use the `std::vector` as that is (from what I understand) our current best practice for this scenario. Is there a particular reason to switch away from that pattern?

Reply:
I agree with you @syntheticmagus, using ```std::vector``` will be the best practice and will not change the final result.

Reply:
Approach has changed, so this is no longer valid.